src/integration.test.ts

```diff
@@ -16,7 +16,7 @@ const getKey = (paths: string[], key: string): string => {
 };
 
 describe("Integration Test", (): void => {
-  const EXEC_OPTIONS = { shell: "/usr/bin/bash" };
+  const EXEC_OPTIONS = { shell: "/bin/bash" };
```

Comment:
I wonder what happens if we stick with `/usr/bin/bash` on Linux but use `C:\Program Files\Git\bin\bash.exe` on Windows. That path to Git Bash is what the GitHub Actions runner uses when `shell: bash` is specified. Keep in mind that you will have to escape the backslashes.

Reply:
It fails the integration test; I get the following output from the integration test:
```
Stack trace: Error: expect(jest.fn()).nthCalledWith(n, ...expected)

    n: 1
    - Expected
    + Received

      "docker image list --format \"{{ .Repository }}:{{ .Tag }}\"",
      Object {
    -   "shell": "C:\\Program Files\\Git\\bin\\bash.exe",
    +   "shell": "/bin/bash",
      },
      [Function anonymous],

    Number of calls: 1

      101 |   ): void => {
      102 |     expect(core.info).nthCalledWith<[string]>(infoCallNum, command);
    > 103 |     expect(child_process.exec).nthCalledWith<[string, ExecOptions, any]>(
          |                                ^
      104 |       execCallNum,
      105 |       command,
      106 |       EXEC_OPTIONS,

      at assertExecBashCommand (integration.test.ts:103:32)
      at assertLoadDockerImages (integration.test.ts:133:7)
```

Reply:
Now I am trying to do branching on Linux and Windows.  When I run the integration test via `yarn test integration.test.ts`, I get:
```
 Stack trace: Error: expect(jest.fn()).nthCalledWith(n, ...expected)

    n: 1
    - Expected
    + Received

      "docker image list --format \"{{ .Repository }}:{{ .Tag }}\"",
      Object {
    -   "shell": "/usr/bin/bash",
    +   "shell": "/bin/bash",
      },
      [Function anonymous],

    Number of calls: 1

      106 |   ): void => {
      107 |     expect(core.info).nthCalledWith<[string]>(infoCallNum, command);
    > 108 |     expect(child_process.exec).nthCalledWith<[string, ExecOptions, any]>(
          |                                ^
      109 |       execCallNum,
      110 |       command,
      111 |       EXEC_OPTIONS,

      at assertExecBashCommand (integration.test.ts:108:32)
      at assertLoadDockerImages (integration.test.ts:138:7)
      at integration.test.ts:216:7
```
I used the following ternary in my attempt to branch on the os:
```
const EXEC_OPTIONS =
    process.platform === "linux"
      ? { shell: "/usr/bin/bash" }
      : { shell: "/bin/bash" };
```
I used console.logs/Console Ninja to confirm that the integration test is running on "linux" and that `EXEC_OPTIONS` is `"/usr/bin/bash"`.  I am a bit befuddled because I thought "/usr/bin/bash" was what it was before we tried to add in MacOS/Windows support and that seemed to work then.  What am I missing?

Reply:
The idea was to try using `C:\\Program Files\\Git\\bin\\bash.exe` on Windows since we already saw that `/bin/bash` didn't work. Are you setting `EXEC_OPTIONS` via branching in the test or the production code? We don't need the tests to pass though just to check whether that's the correct path to Bash in the windows-2022 GitHub Actions runner image.